pkgs/by-name/to/todesk/package.nix

```diff
@@ -0,0 +1,147 @@
+{
+  stdenv,
+  lib,
+  procps,
+  fetchurl,
+  dpkg,
+  writeShellScript,
+  buildFHSEnv,
+  nspr,
+  kmod,
+  systemdMinimal,
+  glib,
+  pulseaudio,
+  libXext,
+  libX11,
+  libXrandr,
+  glibc,
+  cairo,
+  libdrm,
+  coreutils,
+  libXi,
+  libGL,
+  bash,
+  libXcomposite,
+  libXdamage,
+  libXfixes,
+  libXtst,
+  nss,
+  libXxf86vm,
+  gtk3,
+  gdk-pixbuf,
+  pango,
+  libz,
+  libayatana-appindicator,
+}:
+
+let
+  todesk-unwrapped = stdenv.mkDerivation (finalAttrs: {
+    pname = "todesk-unwrapped";
+    version = "4.7.2.0";
+    src = fetchurl {
+      url = "https://newdl.todesk.com/linux/todesk-v${finalAttrs.version}-amd64.deb";
+      hash = "sha256-v7VpXXFVaKI99RpzUWfAc6eE7NHGJeFrNeUTbVuX+yg=";
+      curlOptsList = [
+        "--user-agent"
+        "Mozilla/5.0"
+      ];
+    };
+    nativeBuildInputs = [ dpkg ];
+
+    unpackPhase = ''
+      runHook preUnpack
+      dpkg -x $src ./todesk-src
+      runHook postUnpack
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir "$out"
+      cp -r todesk-src/* "$out"
+      cp "${libayatana-appindicator}/lib/libayatana-appindicator3.so.1" "$out/opt/todesk/bin/libappindicator3.so.1"
+      mv "$out/opt/todesk/bin" "$out/lib"
+      cp -r "$out/lib" "$out/bin"
+      mkdir "$out/opt/todesk/config"
+      mkdir "$out/opt/todesk/bin"
+      chmod +x "$out/lib/libappindicator3.so.1"
+      mkdir "$out/share"
+      mkdir "$out/share/applications"
+      mkdir "$out/share/icons"
+      runHook postInstall
+    '';
+
+  });
+
+in
+buildFHSEnv {
+  name = "todesk";
+  version = "4.7.2.0";
+  targetPkgs = pkgs: [
+    todesk-unwrapped
+    pulseaudio
+    nspr
+    kmod
+    libXi
+    systemdMinimal
+    glib
+    libz
+    bash
+    coreutils
+    libX11
+    libXext
+    libXrandr
+    glibc
+    libdrm
+    libGL
+    procps
+    cairo
+    libXcomposite
+    libXdamage
+    libXfixes
+    libXtst
+    nss
+    libXxf86vm
+    gtk3
+    gdk-pixbuf
+    pango
+
+  ];
+  extraBwrapArgs = [
+    "--bind /var/lib/todesk /opt/todesk/config" # create the folder before bind to avoid permission denided.
+    "--bind ${todesk-unwrapped}/bin /opt/todesk/bin"
+    "--bind /home /home"
+    "--bind /var/lib/todesk /etc/todesk" # service write uuid here. Such a pain!
+  ]; # soft link doesn't work so that we should bind ourselves
+  runScript = writeShellScript "ToDesk.sh" ''
+    export LIBVA_DRIVER_NAME=iHD
+    export LIBVA_DRIVERS_PATH=${todesk-unwrapped}/bin
+    if [ "''${1}" = 'desktop' ]
+    then
+        /opt/todesk/bin/ToDesk
+    elif [ "''${1}" = 'service' ]
+    then
+        /opt/todesk/bin/ToDesk_Service
+    else
+        ''${1}
```

Comment:
Is this necessary?

Reply:
This isn't resolved. Please let me know, why we should be able to run arbitrary commands in our `bubblewrap`environment